File: mediaforge/models/job.py
from enum import StrEnum
from typing import Literal

from pydantic import BaseModel, Field, field_validator


class OutputType(StrEnum):
    main_image = "main_image"
    detail_page = "detail_page"
    video = "video"
    social = "social"


class ImageModelAlias(StrEnum):
    pro = "pro"
    fast = "fast"


class VideoModelAlias(StrEnum):
    veo = "veo"
    seedance = "seedance"


class SkuInput(BaseModel):
    sku_id: str = Field(min_length=1)
    product_image_url: str = Field(min_length=1)
    product_name: str = Field(min_length=1)
    category: str = Field(min_length=1)
    target_platforms: list[str] = Field(default_factory=list)
    output_types: list[str]
    style_hint: str | None = None
    market: str = Field(default="US", min_length=2, max_length=3)
    ref_sku_id: str | None = Field(default=None, description="参考 SKU ID，指定后优先用该 SKU 的图片作为风格参考，未指定时自动 RAG 检索")

    @field_validator("output_types")
    @classmethod
    def _validate_output_types(cls, v: list[str]) -> list[str]:
        allowed = {m.value for m in OutputType}
        for item in v:
            if item not in allowed:
                raise ValueError(f"Invalid output_type: {item}")
        if not v:
            raise ValueError("output_types cannot be empty")
        return v


class AssetOutput(BaseModel):
    sku_id: str
    output_type: str
    file_path: str | None = None
    model_used: str
    platform: str | None = None
    status: Literal["success", "failed", "retrying"]
    error: str | None = None


class BatchSubmitPayload(BaseModel):
    skus: list[SkuInput] = Field(min_length=1, max_length=5000)
    image_model: ImageModelAlias = ImageModelAlias.pro
    video_model: VideoModelAlias = VideoModelAlias.veo
    priority: Literal["low", "normal", "high"] = "normal"

    @property
    def total_skus(self) -> int:
        return len(self.skus)

File: frontend/src/types/job.ts
export type JobStatus = 'pending' | 'running' | 'done' | 'failed' | 'partial_fail'

export type OutputType = 'main_image' | 'detail_page' | 'video' | 'social'
export type ImageModelAlias = 'pro' | 'fast'
export type VideoModelAlias = 'veo' | 'seedance'
export type Priority = 'low' | 'normal' | 'high'

export type Job = {
  job_id: string
  status: JobStatus
  total_skus: number
  done_skus: number
  created_at: string | null
}

export type SkuInput = {
  sku_id: string
  product_image_url: string
  product_name: string
  category: string
  target_platforms: string[]
  output_types: OutputType[]
  style_hint: string | null
  market: string
  ref_sku_id: string | null
}

export type AssetStatus = 'pending' | 'success' | 'failed' | 'retrying'

export type Asset = {
  asset_id: string
  job_id: string
  tenant_id: string
  sku_id: string
  output_type: string
  platform: string | null
  model_used: string
  file_path: string | null
  status: AssetStatus
  error_msg: string | null
  created_at: string
}

export type JobDetail = {
  job_id: string
  status: JobStatus
  total_skus: number
  done_skus: number
  input_data: BatchSubmitPayload | null
  created_at: string | null
  started_at: string | null
  finished_at: string | null
  assets: Asset[]
}

export type BatchSubmitPayload = {
  skus: SkuInput[]
  image_model: ImageModelAlias
  video_model: VideoModelAlias
  priority: Priority
}
